suricata/update/engine.py

```diff
@@ -178,9 +181,12 @@ def test_configuration(suricata_path, suricata_conf=None, rule_filename=None):
 
     logger.debug("Running %s; env=%s", " ".join(test_command), str(env))
     rc = subprocess.Popen(test_command, env=env).wait()
-    if rc == 0:
-        return True
-    return False
+    ret = True if rc == 0 else False
+
+    # Cleanup the temp dir
+    shutil.rmtree(tempdir.name)
```

Comment:
This causes a crash for me as the tempdir tries to remove itself on destruction, but its already gone.  I think `tempdir.cleanup()` will fix this instead of using rmtree.  But again, not available on Python 2 which I'd still like to support where possible.

Reply:
I do not see a `cleanup` function in the instance of tempdir (with `mkdtemp`). Should I just wrap this rmtree call in a try except block to avoid getting a traceback in such cases?

Reply:
> I do not see a `cleanup` function in the instance of tempdir (with `mkdtemp`). Should I just wrap this rmtree call in a try except block to avoid getting a traceback in such cases?

The tempdir.cleanup() only exists on Python 3 with tempfile.TemporaryDirectory().  The crash comes from within the module, not your code.

Best option is to use mkdtemp, and then keep the manual cleanup that you now, as mkdtemp doesn't have cleanup built-in.